### PR TITLE
[#6] Time exceptions too

### LINKED ISSUE #6: Also time actions when they throw an Exception
When the action throws an exception during the use of `timeI`, instrument doesn't send any time metrics about that failure.

### PR DESCRIPTION
**Pull request by korayal**
Closes #6

### CODE REVIEW

File: instrument/src/Instrument/Client.hs [hunk 1]
```diff
@@ -219,6 +223,31 @@ timeI nm hostDimPolicy rawDims i act = do
...
   return res
 
+-- | Run a monadic action while measuring its runtime. Push the measurement into
+-- the instrument system. rethrows exceptions and sends a different Metric on
+-- failure
+--
+-- >>> timeExI \"fileUploadTimeError\" \"fileUploadTime\" policy dims instr $ uploadFile file
+timeExI
+  :: (MonadIO m, MonadCatch m)
+  => MetricName
```
Comment (MichaelXavier) [inline instrument/src/Instrument/Client.hs]: I wonder if we should make this API a bit more flexible. It seems to me that if there's an error, in addition to using a different metric name, we might want to have a different host dimension policy and set of dimensions. What if we changed it so the type was `(Either SomeException a -> (MetricName, HostDimensionPolicy, Dimensions)) -> Instrument -> m a -> m a`?

File: instrument/src/Instrument/Measurement.hs [hunk 2]
```diff
@@ -31,6 +34,15 @@ time act = do
   let !delta = end - start
   return (delta, result)
 
+-- | Measure how long action took, even if they fail
+timeEx :: (MonadCatch m, MonadIO m) => m a -> m (Double, Either SomeException a)
```
Comment (MichaelXavier) [inline instrument/src/Instrument/Measurement.hs:38]: Couldn't this be implemented as `timeEx = time . tryAny`?

**Review by MichaelXavier — CHANGES_REQUESTED** [review]

File: instrument/src/Instrument/Measurement.hs (hunk 2 above)
Comment (korayal) [reply instrument/src/Instrument/Measurement.hs:38]: Yup, that makes much more sense. Thanks!

File: instrument/src/Instrument/Client.hs (hunk 1 above)
Comment (korayal) [reply instrument/src/Instrument/Client.hs]: Yes, at the very least it doesn't confuse the caller which one is which. updated.

**Comment by korayal** [comment]
@MichaelXavier Since this was failing on Stackage Nightly, I've dropped the CI checks for versions lower than 8.6.5.

**Pull request merged**